docs/data/charts/styling/styling.md

```diff
@@ -42,6 +42,67 @@ Here is an example of the d3 Categorical color palette.
 
 {{"demo": "ColorTemplate.js"}}
 
+### Values color
+
+Colors can also be set according to item values.
+To do so, use the `colorMap` property of the corresponding axis.
+
+For demonstration, you can refer to the chart specific pages:
+
+- [bar chart](/x/react-charts/bars/#color-scale)
+- [line chart](/x/react-charts/lines/#color-scale)
+- [scatter chart](/x/react-charts/scatter/#color-scale)
+
+The `colorMap` property can accept three kind of objects defined bellow
+
+#### Piecewise color map
+
+The piecewise configuration takes an array of N `thresholds` values, and N+1 `colors`.
+
+```ts
+{
+  type: 'piecewise';
+  thresholds: V[];
+  colors: string[];
+}
+```
+
+#### Continuous color map
+
+The continuous configuration allows to map values from `min` to `max` properties, to it's color.
+
+The `color` property can either be an array of two colors to interpolate.
+Or directly an interpolation function that for a number `t` between 0 and 1 returns a color.
+The [d3-scale-chromatic](https://d3js.org/d3-scale-chromatic) offers a lot of those functions.
+
+```ts
+{
```

Comment:
What happens to the value outside of the [min, max] range? Can you define several continuous scales to cover the whole dataset?

Reply:
Outside the range it uses the values for 0 or 1 (depends on which side you overflow)

For example in [scatter chart demo](https://deploy-preview-12490--material-ui-x.netlify.app/x/react-charts/scatter/#color-scale) the continuous scale goes from -2 to 2, but some data point are outside this range

Here is a screen shot with more emphasis (min/max set to -1/1)

![image](https://github.com/mui/mui-x/assets/45398769/bf8b33b2-5c8f-4053-89d7-3d514639f5f6)


This idea of not usingthe data set range come from Echarts. I was first surprised, but it has two benefits:
- easier to implement 🦥
- enforce devs to have a stable meaning full range.

For example temperature, you could say we use data set min/max, but does nto make sens If you compare a plot of temperature in Lille or in Sahara. The first one will be full red at 10°C the last will be red at 50°C so the color lose lots of its meaning

Reply:
OK, that makes a lot of sense
Maybe we could develop slightly more in the doc :+1: